### PR TITLE
Hw7 task1

### PR DESCRIPTION
**Pull request by E1-Bart0**

### CODE REVIEW

File: All_home_works/hw7/task1.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    str, list, tuple, dict, set, int, bool
+"""
+from typing import Any
+
+
+def find_occurrences(tree: dict, element: Any) -> int:
+    counter = 0
+    if tree == element and type(tree) == type(element):
+        counter += 1
+    elif isinstance(tree, tuple) or isinstance(tree, list) or isinstance(tree, set):
```
Comment (Cjkjvfnby) [inline All_home_works/hw7/task1.py]: We can simplify this line. Please check the documentation.

File: test/hw7/test_task1.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            ["OK"],
+            2,
+        ),
+    ],
+)
+def test_find_occurrences__search__list__tuple__set(input_data, element, expected):
+    assert expected == find_occurrences(input_data, element)
+
+
+@pytest.mark.parametrize(  # noqa: PT006
```
Comment (Cjkjvfnby) [inline test/hw7/test_task1.py]: What is PT006 about?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Looks good, couple of minor comments.

File: All_home_works/hw7/task1.py (hunk 1 above)
Comment (E1-Bart0) [reply All_home_works/hw7/task1.py]: `isinstance(tree, (list, tuple, set))`

File: test/hw7/test_task1.py (hunk 2 above)
Comment (E1-Bart0) [reply test/hw7/test_task1.py]: Thanks for asking. Maybe you will help me to sort out.
`@pytest.mark.parametrize("data,expected", [("a", "a"), ("b", "b")])`
GOT: PT006 wrong name(s) type expected tuple.

`@pytest.mark.parametrize("data,expected", (("a", "a"), ("b", "b")))`
GOT: PT007 wrong values expected list of tuples
AND: PT006 wrong name(s) type expected tuple

`@pytest.mark.parametrize("data,expected", (["a", "a"], ["b", "b"]))`
GOT: PT006 wrong name(s) type expected tuple, 
AND: PT007 wrong values type expected list of tuples

Comment (Cjkjvfnby) [reply test/hw7/test_task1.py]: `PT006` is an error code that is produced by flake8 (actually by one of its plugins).  This path will lead you to the plugin page and after that, you will meet the page dedicated to this code.  You will find good examples there. 

This approach usually works well except S??? codes. In that case, you need to search for the B??? codes. (S -> security in the bandit-flake8-plugin, B -> is in the bandit tools that used by plugin)

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: test/hw7/test_task1.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            2,
+        ),
+    ],
+)
+def test_find_occurrences__search__list__tuple__set(input_data, element, expected):
+    assert expected == find_occurrences(input_data, element)
+
+
+@pytest.mark.parametrize(
+    ("input_data", "element", "expected"),
```
Comment (Cjkjvfnby) [inline test/hw7/test_task1.py:57]: If expected is always 1, why do you need it?

File: test/hw7/test_task1.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        ({"key1": 1, "key2": True}, True, 1),
+        ({"key1": [True], "key2": ["NOT OK", False]}, True, 1),
+        ({"key1": {"key1": ["OK", True]}, "key2": {"key1": ("OK", None)}}, False, 0),
+    ],
+)
+def test_find_occurrences__search_bool(input_data, element, expected):
+    assert expected == find_occurrences(input_data, element)
+
+
+def test_find_occurrences__many_occurrences__example_from_description():
```
Comment (Cjkjvfnby) [inline test/hw7/test_task1.py]: I'd prefer to split it into different tests, each for a specific test.

File: All_home_works/hw7/task1.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    for new_element in array:
+        counter += find_occurrences(new_element, element)
+    return counter
+
+
+def find_occurrences_in_dict(dictionary, element):
+    counter = 0
+    for key, value in dictionary.items():
+        counter += find_occurrences(key, element)
+        counter += find_occurrences(value, element)
```
Comment (Cjkjvfnby) [inline All_home_works/hw7/task1.py:33]: There is not test with a name that says that you test occurrence in key and value.  So probably this code is not tested.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**